mizar/dp/mizar/operators/droplets/droplets_operator.py

```diff
@@ -99,6 +99,13 @@ def assign_divider_droplet(self, divider):
         d = random.sample(droplets, 1)[0]
         divider.set_droplet(d)
 
+    def assign_ftn_droplet(self, ftn):
+        droplets = set(self.store.get_all_droplets())
+        for d in droplets:
+            if not d.known_ftn:
+                ftn.set_droplet(d)
+                break
```

Comment:
How about moving this search loop to the store? Basically introduce a function get_non_ftn_droplets from the store to get a set of droplets that are not ftns. The store droplet apis need to maintain this list and return them in __constant time__. Then the assignment algorithm here selects one of them (random for now).